src/NServiceBus.Persistence.CosmosDB.TransactionalSession.AcceptanceTests/ConfigureEndpointCosmosDBPersistence.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Extensions.DependencyInjection;
+using NServiceBus;
+using NServiceBus.AcceptanceTesting;
+using NServiceBus.AcceptanceTesting.Support;
+using NServiceBus.Persistence.CosmosDB;
+using NServiceBus.TransactionalSession;
+using NServiceBus.TransactionalSession.AcceptanceTests;
+
+public class ConfigureEndpointCosmosDBPersistence : IConfigureEndpointTestExecution
+{
+    public Task Configure(string endpointName, EndpointConfiguration configuration, RunSettings settings, PublisherMetadata publisherMetadata)
+    {
+        var persistence = configuration.UsePersistence<CosmosPersistence>();
+        persistence.DisableContainerCreation();
+        persistence.CosmosClient(SetupFixture.CosmosDbClient);
+        persistence.DatabaseName(SetupFixture.DatabaseName);
+        persistence.EnableTransactionalSession();
+
+        configuration.RegisterComponents(services => services.AddSingleton<IPartitionKeyFromHeadersExtractor, PartitionKeyProvider>());
+        configuration.RegisterComponents(services =>
+            services.AddSingleton<IContainerInformationFromHeadersExtractor, ContainerInformationProvider>());
+
+        return Task.CompletedTask;
+    }
+
+    public Task Cleanup() => Task.CompletedTask;
+
+    class PartitionKeyProvider : IPartitionKeyFromHeadersExtractor
```

Comment:
should the tx session tests really use these default providers? Given the user has to specify the partition key on the options, this shouldn't be needed?

Reply:
On the incoming side it still requires a mapping for the actual messages that are delivered. All ATTs in this project are aligned to use the test run id as the partition key and a default container unless otherwise specified so I aligned that. The benefit is there is no mapping needed for the incoming messages in the actual tests